lib/webmock/util/headers.rb

```diff
@@ -57,6 +60,23 @@ def self.basic_auth_header(*credentials)
         "Basic #{Base64.strict_encode64(credentials.join(':')).chomp}"
       end
 
+      def self.normalize_name(name)
+        name
+          .to_s
+          # By using `tr` here we can normalize the headers ahead of time
+          .tr(NONSTANDARD_HEADER_DELIMITER, STANDARD_HEADER_DELIMITER)
+          # ...which will be faster because `split` on a `String` is a good deal faster
+          # than on a `Regexp`.
+          .split(STANDARD_HEADER_DELIMITER)
+          # Now since this is state we control here we can mutate it with `map!` to again
+          # avoid extra object creations.
+          .map!(&:capitalize)
+          # Then join it back together. Most of these constants are to avoid recreating
+          # these strings hundreds of thousands of times in larger codebases. Ours I
+          # believe was in the millions.
+          .join(STANDARD_HEADER_DELIMITER)
```

Comment:
I may remove these comments real quick, as they don't add a lot of value to the library itself.